kiosk/src/Components/HighScoresList.tsx

```diff
@@ -3,63 +3,45 @@ import { HighScore } from "../Models/HighScore";
 import { Kiosk } from "../Models/Kiosk";
 
 interface IProps {
-    kiosk: Kiosk
+    kiosk: Kiosk;
+    gameId: string;
+    highScoreMode: string;
   }
 
-const HighScoresList: React.FC<IProps> = ({ kiosk }) => {
-    const [selectedGameId, setSelectedGameId] = useState(kiosk.selectedGame?.id);
-
-    const renderList = (): JSX.Element[] => {
-        const highScores: HighScore[] = kiosk.getHighScores(selectedGameId!);
-
-        if (!highScores || !highScores.length) {
-            return ([
-                <tr key="noscores">
-                    <td className="highScoreSmallDiv">None yet</td>
-                </tr>
-            ]);
-        }
-
-        return highScores.slice(0, 3).map((highScore, index) => {
-            return (
-                <tr className="highScoreSmallDiv" key={index}>
-                    <td className="highScoreSmallIndex">{index + 1}.</td>
-                    <td className="highScoreSmallInitials">
-                        {highScore.initials}
-                    </td>
-                    <td className="highScoreSmallInitialsScoreSpacer">&nbsp;</td>
-                    <td className="highScoreSmallScore">{highScore.score}</td>
-                </tr>
-            );
-        })
-    }
-
-    useEffect(() => {
-        kiosk.onGameSelected = () => {
-            setSelectedGameId(kiosk.selectedGame!.id);
-          };
-    });
-
-    if (!selectedGameId) {
-        return (
-            <span className="center">
-                No game selected.
-            </span>
-        );
-    }
-
-    if (!kiosk.selectedGame || (kiosk.selectedGame!.highScoreMode === "None")) {
-        return (<></>);
-    }
+const HighScoresList: React.FC<IProps> = ({ kiosk, gameId, highScoreMode }) => {
+    const highScores: HighScore[] = kiosk.getHighScores(gameId);
```

Comment:
nit; not a 'regression' from this pr but something that'd be nice to clean up in a future one, if we're only using this prop to grab the high scores, it'd be nice to pass just those high scores in from `gamelist.tsx` to gameslide / here -- it can be hard to wrap your head around / refactor things when you have to get into the context of 'I have this object from my parents parent that I have to call a method on to get x value', and over time can make it  -- we have a big case of this problem in our `IProjectView` in the main app (which is basically just `app.tsx` and gets passed around a bunch). 

this is also the case if you just want a few functions or handlers -- if you need to view/edit high scores, passing in a pair of `set highscore` and `get highscore`  handlers can make the code easier to follow as you only have to see what's relevant within the score of this component, and the parent elements can determine how that is completed without exposing exactly how it's done / making it so other components take dependencies on implementation details.

Reply:
Thanks for pointing this out! I don't think this is hard to implement and goes hand-in-hand with the changes that I made, so I'll update this.